v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/PostgreSQLJdbcValueMappings.java

```diff
@@ -19,34 +19,216 @@
+      };
+
+  private static final ResultSetValueExtractor<OffsetTime> timetzExtractor =
+      (rs, fieldName) -> {
+        String timeTz = rs.getString(fieldName);
+        if (timeTz == null) {
+          return null;
+        }
+        return OffsetTime.parse(timeTz, TIMETZ_FORMAT);
+      };
+
+  private static final ResultSetValueExtractor<java.sql.Timestamp> timestampExtractor =
+      (rs, fieldName) -> rs.getTimestamp(fieldName, UTC_CALENDAR);
+
+  private static final ResultSetValueExtractor<OffsetDateTime> timestamptzExtractor =
+      (rs, fieldName) -> {
+        String timestampTz = rs.getString(fieldName);
+        if (timestampTz == null) {
+          return null;
+        }
+        return OffsetDateTime.parse(timestampTz, TIMESTAMPTZ_FORMAT);
+      };
+
+  // Value might be a Double.NaN or a valid BigDecimal
+  private static final ResultSetValueMapper<Number> numericToAvro =
+      (value, schema) -> value.toString();
+
+  private static final ResultSetValueMapper<java.sql.Date> dateToAvro =
       (value, schema) -> (int) value.toLocalDate().toEpochDay();
 
-  // TODO(thiagotnunes): Add missing type mappings
+  private static final ResultSetValueMapper<java.sql.Timestamp> timestampToAvro =
+      (value, schema) -> toMicros(value.toInstant());
+
+  private static final ResultSetValueMapper<OffsetTime> timeToAvro =
+      (value, schema) -> toMicros(value);
+
+  private static final ResultSetValueMapper<OffsetTime> timetzToAvro =
+      (value, schema) ->
+          new GenericRecordBuilder(TimeTz.SCHEMA)
+              .set(
+                  TimeTz.TIME_FIELD_NAME,
+                  TimeUnit.NANOSECONDS.toMicros(value.toLocalTime().toNanoOfDay()))
+              .set(
+                  TimeTz.OFFSET_FIELD_NAME,
+                  TimeUnit.SECONDS.toMillis(value.getOffset().getTotalSeconds()))
+              .build();
+
+  private static final ResultSetValueMapper<OffsetDateTime> timestamptzToAvro =
+      (value, schema) ->
+          new GenericRecordBuilder(TimeStampTz.SCHEMA)
+              .set(TimeStampTz.TIMESTAMP_FIELD_NAME, toMicros(value.toInstant()))
+              .set(
+                  TimeStampTz.OFFSET_FIELD_NAME,
+                  TimeUnit.SECONDS.toMillis(value.getOffset().getTotalSeconds()))
+              .build();
+
   private static final ImmutableMap<String, JdbcValueMapper<?>> SCHEMA_MAPPINGS =
       ImmutableMap.<String, Pair<ResultSetValueExtractor<?>, ResultSetValueMapper<?>>>builder()
           .put("BIGINT", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("BIGSERIAL", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("BIT", Pair.of(bytesExtractor, valuePassThrough))
+          .put("BIT VARYING", Pair.of(bytesExtractor, valuePassThrough))
+          .put("BOOL", Pair.of(ResultSet::getBoolean, valuePassThrough))
+          .put("BOOLEAN", Pair.of(ResultSet::getBoolean, valuePassThrough))
+          .put("BOX", Pair.of(ResultSet::getString, valuePassThrough))
```

Comment:
For MySQL, types that are not supported by SMT for schema migration are marked unsupported to fail fast. 
If we have made a conscious choice to support the Geometric types with strings, It's fine to have this change (for schema migration other things come into consideration, especially around, if the user has an index mounted on these special types, would converting them to a string be ok?)

Reply:
I tried following this https://cloud.google.com/datastream/docs/unified-types#map-psql for type mapping, so you will see that the `PostgreSQLMappingProvider.java` has some types marked as unsupported. Indexes on geometric types using Gist which is not supported by Spanner.

What do you suggest?

Reply:
Got it.
I think to have parity with live, we should map unsupported types to unsupported.
In which case, the schema will always be a NULL, and so we might not need a `value mapping` for those types until we support them.

Reply:
Done